src/ansys/sound/core/sound_composer/source_audio.py

```diff
@@ -0,0 +1,207 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Sound Composer's audio source."""
+import warnings
+
+from ansys.dpf.core import Field, Operator
+from matplotlib import pyplot as plt
+import numpy as np
+from numpy import typing as npt
+
+from ansys.sound.core.signal_utilities import LoadWav, Resample
+
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+from ._source_parent import SourceParent
+
+ID_LOAD_FROM_TEXT = "load_sound_samples_from_txt"
+
+
+class SourceAudio(SourceParent):
+    """Sound Composer's audio source class.
+
+    This class creates an audio source for the Sound Composer.
+    """
+
+    def __init__(self, file: str = ""):
+        """Class instantiation takes the following parameters.
+
+        Parameters
+        ----------
+        file : str, default ""
+            Path to the audio file (WAV format).
+        """
+        super().__init__()
+
+        # Load the audio file, if specified.
+        if len(file) > 0:
+            loader = LoadWav(file)
+            loader.process()
+            self.source_audio_data = loader.get_output()[0]
+        else:
+            self.source_audio_data = None
+
+        # Define DPF Sound operators.
+        self.__operator_load = Operator(ID_LOAD_FROM_TEXT)
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if self.source_audio_data is not None:
+            support_data = self.source_audio_data.time_freq_support.time_frequencies.data
+
+            if len(support_data) < 1:
+                str_duration = "N/A"
+            else:
+                str_duration = f"{support_data[-1]:.1f} s"
+
+            if len(support_data) < 2:
+                str_fs = "N/A"
+            else:
+                str_fs = f"{1/(support_data[1] - support_data[0]):.1f} Hz"
+
+            str_source = (
+                f"'{self.source_audio_data.name}'\n"
+                f"\tDuration: {str_duration}\n"
+                f"\tSampling frequency: {str_fs}"
+            )
+        else:
+            str_source = "Not set"
+
+        return f"Audio source: {str_source}"
+
+    @property
+    def source_audio_data(self) -> Field:
+        """Audio source data, as a DPF field.
+
+        Audio samples over time, in Pa, as a DPF field.
+        """
+        return self.__source_audio_data
+
+    @source_audio_data.setter
+    def source_audio_data(self, source_audio_data: Field):
+        """Set the audio source data, from a DPF field."""
+        if not (isinstance(source_audio_data, Field) or source_audio_data is None):
+            raise PyAnsysSoundException(
+                "Specified audio source data must be provided as a DPF field."
+            )
+        self.__source_audio_data = source_audio_data
+
+    def load_source_audio_from_text(self, file: str):
```

Comment:
@ansaminard maybe the spec was not clear on our side so apologies for that but, shouldn't we simply deal with loading text files in the constructor, the same way we load wav files ? 

Based on the file's extension we use one operator or the other ?

Reply:
@a-bouth It is a bit like in SourceSpectrum (with discussed with @anshlachamb at the time). I'm not too fond of calling DPF operators already in the ctor.
Plus, this way, the user also has access to this method and can choose (for whatever reason) to first instantiate the class and then load the file (whether wav or txt)

Reply:
@ansaminard yes but there's no method `load_source_audio_from_wav` in your class, so either the user specifies the **wav** in the constructor or the **text** in a separate method... I don't find this very homogenous. 

What is bothering you regarding "calling DPF Operators already in the ctor" ? Technically you're already doing it by using `LoadWav` so might as well use a second one ?

Reply:
@a-bouth I got confused with SourceControlTime where I have both methods. But you're right to point out than I'm merely filling the gaps in the specifications. This operator was mentioned (as optional), but how it was supposed to be used was not indicated. So as I said I discussed this a bit with @anshlachamb, and I recall she was ok with that.
You are also right that it's not too consistent to do things differently for wav and txt, but I would tend to move the call to LoadWav into its own method rather than removing that of the call to load_source_audio_from_wav 
And as for why I don't like calling things in the ctor, I _am_ calling the DPF operator in the ctor, it's just that this is done by calling another function. I feel it is more "unitary" and easier to read

Reply:
@ansaminard okay, I think I get your point of view. 

Then, would it be fine with you if : 
* You move the `LoadWav` in its own function for this class (most likely `load_source_audio_from_wav` 
* The user can specify either a text or wav file in the CTOR and based on the extension you then call the appropriate `load_source_audio_from_*`?

Reply:
@a-bouth Agreed.

Reply:
@a-bouth Done